app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save{ self.email = email.downcase }
+  validates :name, presence: true, length:{ maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :email, presence: true, length:{ maximum: 255 },
+   format:{ with: VALID_EMAIL_REGEX }, 
+   uniqueness:{ case_sensitive: false }
+  has_secure_password
+  validates :password, presence: true, length:{ minimum: 6 }
```

Comment:
In this file, you can see many constants. Please use "Config" gem to manage the constants 
Reference: [https://github.com/railsconfig/config](https://github.com/railsconfig/config)